### PR TITLE
*: share etcd client from domain for autoid allocator

### LINKED ISSUE #46324: etcd client leak when a table uses AUTO_ID_CACHE=1 and auto_increment
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
nowhere close this etcd client
https://github.com/pingcap/tidb/blob/1769f3a1ac3674b9267dca55bf1fc261ac804325/meta/autoid/autoid.go#L588

### 4. What is your TiDB version? (Required)
this code is introduced in tidb version >= 6.4.0

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #46324

Problem Summary:

The autoid allocate service need a etcd client, we can create one and close it after use.
But every time infoschema reload, we need to create one...
A better solution is to share the etcd client in domain.Domain (also ddl.DDL)

### What is changed and how it works?

Refactor the auto id parameter from `kv.Store` to `autoid.Requirement`
The `autoid.Requirement` now consists of a kv.Store and a clientv3.Client...
This change affect quite a lot packages, especially tools packages.

### Check List

Tests 

This is a code refactor so all the old tests should not be break.

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lcwangchao — APPROVED** [review]

File: br/pkg/lightning/importer/table_import.go [hunk 1]
```diff
@@ -316,6 +320,21 @@ func (tr *TableImporter) populateChunks(ctx context.Context, rc *Controller, cp
 	return err
 }
 
+type asAutoIDRequirement TableImporter
+
+func (r *asAutoIDRequirement) Store() tidbkv.Storage {
```
Comment (D3Hunter) [inline br/pkg/lightning/importer/table_import.go:325]: impl those methods on `*TableImporter` directly?

**Review by D3Hunter — COMMENTED** [review]
lightning/import part lgtm

File: br/pkg/lightning/importer/table_import.go (hunk 1 above)
Comment (tiancaiamao) [reply br/pkg/lightning/importer/table_import.go]: Done

**Review by GMHDBJD — APPROVED** [review]
lgtm

**Review by tangenta — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]
br part LGTM

File: pkg/meta/autoid/autoid.go [hunk 2]
```diff
@@ -568,41 +566,20 @@ func NextStep(curStep int64, consumeDur time.Duration) int64 {
...
 		isUnsigned: isUnsigned,
 		keyspaceID: keyspaceID,
 	}
-	if len(addrs) > 0 {
-		etcdCli, err := clientv3.New(clientv3.Config{
-			Endpoints:        addrs,
-			AutoSyncInterval: 30 * time.Second,
-			TLS:              ebd.TLSConfig(),
-		})
-		etcd.SetEtcdCliByNamespace(etcdCli, keyspace.MakeKeyspaceEtcdNamespaceSlash(store.GetCodec()))
```
Comment (moonsphere) [inline pkg/meta/autoid/autoid.go:597]: It removed the autoid client's own `SetEtcdCliByNamespace(cli,
  MakeKeyspaceEtcdNamespaceSlash(...))` and switched to using the domain's shared etcdClient, which was set up with `MakeKeyspaceEtcdNamespace`
  (without trailing slash). This inadvertently reverted the namespace prefix fix from 3203c38477, causing a mismatch between the server (still using slash) and the client (now without slash) when using api v2 or keyspace.

**Pull request merged**